### PR TITLE
Pre-compute group-level average metrics

### LINKED ISSUE #123: Add data summarizing group-level aggregate stats
#84 and #101 involve comparisons between individual companies and aggregate company groups (S&P 500, etc).  Since those data are static, it'd be most efficient to incorporate this into the rest of the data generation process.

### PR DESCRIPTION
**Pull request by jmelot**
Closes #123

### CODE REVIEW

**Review by brianlove — APPROVED** [review]
There's a mismatch in the test files between group key and group name, but otherwise it looks good.

File: web/scripts/retrieve_data.py [hunk 1]
```diff
@@ -75,10 +75,9 @@
 CURRENT_YEAR = _curr_time.year if _curr_time.month > 6 else _curr_time.year - 1
 YEARS = list(range(CURRENT_YEAR - 10, CURRENT_YEAR + 1))
 
-GROUP_NAMES_TO_KEYS = {
-    "S&P 500": "sp500",
-    "Fortune Global 500": "global500"
-}
+# Used (along with a check that we never actually meet or exceed this number with legitimate CSET ids)
+# when creating fake CSET ids for groups
+GROUP_OFFSET = 1_000_000
```
Comment (brianlove) [inline web/scripts/retrieve_data.py:80]: Could this get added to the `overall_data.json` file?  That would let the company/group column formatter function recognize when an entry is a group without risking breaking things in the future.

File: web/tests/test_data/get_average_group_data_input.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
+  "global500": {
+    "cset_id": 1000500,
+    "name": "S&P 500",
```
[the remark this answers is not in the document]
Comment (jmelot) [reply web/tests/test_data/get_average_group_data_input.json:4]: siiigh, fixed

File: web/scripts/retrieve_data.py (hunk 1 above)
Comment (jmelot) [reply web/scripts/retrieve_data.py:79]: Done, I went ahead and integrated it into table_columns.js

**Review by brianlove — APPROVED** [review]
👍

**Pull request merged**